__docs__/wonder-blocks-floating/floating.stories.tsx

```diff
@@ -0,0 +1,232 @@
+import {border, semanticColor, sizing} from "@khanacademy/wonder-blocks-tokens";
+import {BodyText} from "@khanacademy/wonder-blocks-typography";
+
+import packageConfig from "../../packages/wonder-blocks-floating/package.json";
+import ComponentInfo from "../components/component-info";
+import {longText} from "../components/text-for-testing";
+
+type StoryComponentType = StoryObj<typeof Floating>;
+
+export default {
+    title: "Packages / Floating / Floating",
+    component: Floating,
+    parameters: {
+        componentSubtitle: (
+            <ComponentInfo
+                name={packageConfig.name}
+                version={packageConfig.version}
+            />
+        ),
+        chromatic: {
+            // See floating-testing-snapshots.stories.tsx for the snapshot
+            // stories for the different middlewares.
+            disableSnapshot: true,
+        },
+    },
+    decorators: [
+        (Story): React.ReactElement => (
+            <View style={styles.storyCanvas}>{Story()}</View>
+        ),
+    ],
+    args: {
+        content: (
+            <View style={{padding: sizing.size_080}}>Floating content</View>
+        ),
+    },
+} as Meta<typeof Floating>;
+
+/**
+ * Dhis is the most minimal example for Floating. In this case, we use it to
+ * launch a popup that is triggered when the button is clicked.
+ */
+export const Default: StoryComponentType = {
+    render: function Render(args) {
+        const [isOpen, setIsOpen] = React.useState(args.open ?? true);
+
+        React.useEffect(() => {
+            if (args.open !== undefined) {
+                setIsOpen(args.open);
+            }
+        }, [args.open]);
+
+        const buttonLabel = isOpen ? "Close" : "Open";
+
+        return (
+            <Floating {...args} open={isOpen} onOpenChange={setIsOpen}>
+                <Button
+                    kind="primary"
+                    onClick={() => setIsOpen((state) => !state)}
+                >
+                    {buttonLabel}
+                </Button>
+            </Floating>
+        );
+    },
+};
+
+function SwitchWithLabel({
+    checked,
+    onChange,
+    label,
+}: {
+    checked: boolean;
+    onChange: (checked: boolean) => void;
+    label: string;
+}) {
+    const id = React.useId();
+    return (
+        <View style={styles.row}>
+            <Switch checked={checked} onChange={onChange} id={id} />
+            <BodyText tag="label" htmlFor={id}>
+                {label}
+            </BodyText>
+        </View>
+    );
+}
+
+/**
+ * Floating relies on floating-ui's middlewares, which can be configured to
+ * control the positioning and behavior of the floating element:
+ *
+ * - `shift`: Shift the floating element along the axis to keep it in view.
+ * - `flip`: Flip the floating element to the opposite side if there's not enough space.
+ * - `offset`: Offset the floating element from the reference element.
+ * - `hide`: Hide the floating element when the reference element is hidden.
```

Comment:
In the example, I'm unable to scroll in a way where the ref element is not shown, so it is hard to try out the `hide` functionality!